server/src/main/java/org/elasticsearch/search/aggregations/metrics/max/MaxAggregator.java

```diff
@@ -69,7 +74,14 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             final LeafBucketCollector sub) throws IOException {
         if (valuesSource == null) {
             return LeafBucketCollector.NO_OP_COLLECTOR;
-    }
+        }
+        Number segMax = maxFunc.apply(ctx.reader());
+        if (segMax != null) {
+            double max = maxes.get(0);
+            max = Math.max(max, segMax.doubleValue());
+            maxes.set(0, max);
+            throw new CollectionTerminatedException();
```

Comment:
how do we know it's ok to throw this exception and that we won't terminate parent aggregators too?

Reply:
currently this is checked in the factory when we create the function which will return `null` if the aggregator has a parent.